moveit_core/collision_detection/include/moveit/collision_detection/world.h

```diff
@@ -36,13 +36,15 @@
 
 #pragma once
 
-#include <moveit/macros/class_forward.h>
+#include <map>
 #include <string>
 #include <vector>
-#include <map>
-#include <boost/function.hpp>
+
 #include <Eigen/Geometry>
+#include <boost/function.hpp>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
```

Comment:
I'm not really happy about sorting third-party includes by alphabet (at least ROS packages should in theory go after system libs), but that's annoying to achieve. I guess separate including groups for our moveit-specific libs (geometric_shapes, ...), as well as Eigen and boost (possibly also fcl and OMPL) might suffice for 95% of the cases?

Reply:
What do you think of my latest revision, I tried to accomplish what you asked as best I could.  I can't tell what is/is-not a ros library (other than maybe rclcpp) so instead, I just made a block for our external ones and another block for the libraries you listed.